astropy/stats/biweight.py

```diff
@@ -17,7 +28,10 @@
 ]
 
 
-def _stat_functions(data, ignore_nan=False):
+def _stat_functions(
+    data: ArrayLike,
+    ignore_nan: bool | None = False,
+) -> tuple[Callable, Callable]:
```

Comment:
Can you fill out the `Callable` info a bit further.
I don't think this requires a `Protocol.__call__`, but if that's the case then instead match any input using `Callable[..., NDArray[np.floating]]`  (with the correct output) and flag it for a future PR.

Reply:
Yeah it does kind of messy without making a custom callback. Fixed and left a TODO in [882750c](https://github.com/astropy/astropy/pull/16562/commits/882750cf79d48fa93538e26c6ae2298c09bc4ee1)